src/internal/internal_hettmqr.cc

```diff
@@ -278,48 +304,32 @@ void hettmqr(
                         tile::deepConjTranspose(C(j, i1));
                     }
                 }
-                else if (C.tileIsLocal(i2, j)) {
-                    // Sends updated tile back.
-                    int src = (i1 >= j
-                              ? C.tileRank(i1, j)
-                              : C.tileRank(j, i1));
-                    // Send updated tile back.
-                    C.tileSend(i1, j, src, tag);
-                    if (call_tile_tick) {
-                        C.tileTick(i1, j);
-                    }
-                }
             } // for j
-        } // for index
-
-        //--------------------
-        // Finish updating all rows before updating columns.
-        slate_mpi_call(
-            MPI_Barrier(C.mpiComm()));
+        }
+        #pragma omp taskwait
```

Comment:
I guess by moving the data movement up and then sync the tasks, you can get rid of the `MPI_Barrier` call.

Reply:
Actually, I don't think the `MPI_Barrier` call was necessary to begin with since each process only needs to keep its local data & send/recvs consistent.